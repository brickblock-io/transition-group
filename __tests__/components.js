import createApp from '../__test-helpers__/createApp'
import allProps from '../__test-helpers__/allProps'


jest.mock('../src/dom-utils', () => ({
  addClass: (component, className, name) => {
    const log = { method: 'addClass', className, name }
    console.log(log)
    expect(log).toMatchSnapshot()
  },
  removeAnimationClasses: (component, name) => {
    const log = { method: 'removeAnimationClasses', name }
    console.log(log)
    expect(log).toMatchSnapshot()
  },
  setTimeoutAnimationFrame: (func, ms, className) => {
    const log = { method: 'setTimeoutAnimationFrame', ms, className }
    console.log(log)
    expect(log).toMatchSnapshot()

    func()
  },
}))


describe('<TransitionGroup /> + <Transition /> -- no props', () => {
  it('appear', () => {
    const app = createApp()

    console.log(app.tree())
    console.log(app.tree().children[0])

    app.snap()

    return app.story()
  })

  it('enter + leave', () => {
    const app = createApp()

    app.snap({ type: 'SWITCH' })

    console.log(app.tree())
    console.log(app.tree().children[0])

    return app.story()
  })

  it('enter + leave -- simple duration (debounced)', () => {
    const app = createApp({ duration: 500 })

    app.snap({ type: 'SWITCH' })

    console.log(app.tree())
    console.log(app.tree().children[0])

    return app.story()
  })
})


describe('<TransitionGroup /> -- all props filled', () => {
  it('appear', () => {
    const onAppear = jest.fn()
    const app = createApp(allProps({ onAppear }))

    console.log(app.tree())
    console.log(app.tree().children[0])

    expect(onAppear).toBeCalled()

    app.snap()

    return app.story()
  })

  it('beforeAppear', () => {
    const onBeforeAppear = jest.fn()
    const app = createApp(allProps({ onBeforeAppear }))

    console.log(app.tree())
    console.log(app.tree().children[0])

    expect(onBeforeAppear).toBeCalled()

    app.snap()

    return app.story()
  })

  it('enter', () => {
    const onEnter = jest.fn()
    const app = createApp(allProps({ onEnter }), null, { type: 'EMPTY' })

    app.snap({ type: 'FULL' })

    console.log(app.tree())
    console.log(app.tree().children[0])
    expect(onEnter).toBeCalled()

    return app.story()
  })

  it('beforeEnter', () => {
    const onBeforeEnter = jest.fn()
    const app = createApp(allProps({ onBeforeEnter }), null, { type: 'EMPTY' })

    app.snap({ type: 'FULL' })

    console.log(app.tree())
    console.log(app.tree().children[0])
    expect(onBeforeEnter).toBeCalled()

    return app.story()
  })

  it('leave', () => {
    const onLeave = jest.fn()
    const app = createApp(allProps({ onLeave }))

    app.snap({ type: 'EMPTY' })

    console.log(app.tree())
    expect(onLeave).toBeCalled()

    return app.story()
  })

  it('beforeLeave', () => {
    const onBeforeLeave = jest.fn()
    const app = createApp(allProps({ onBeforeLeave }))

    app.snap({ type: 'EMPTY' })

    console.log(app.tree())
    expect(onBeforeLeave).toBeCalled()

    return app.story()
  })

  it('onFull callback', () => {
    const onFull = jest.fn()
    const app = createApp({ onFull }, null, { type: 'EMPTY' })

    console.log(app.tree())
    app.snap({ type: 'FULL' })

    console.log(app.tree())
    expect(onFull).toBeCalled()

    return app.story()
  })

  it('onEmpty callback', () => {
    const onEmpty = jest.fn()
    const app = createApp({ onEmpty })

    console.log(app.tree())
    app.snap({ type: 'EMPTY' })

    console.log(app.tree())
    expect(onEmpty).toBeCalled()

    return app.story()
  })
})


describe('<Transition /> -- all props filled', () => {
  it('appear', () => {
    const onAppear = jest.fn()
    const app = createApp({}, allProps({ onAppear }))

    console.log(app.tree())
    console.log(app.tree().children[0])

    expect(onAppear).toBeCalled()

    app.snap()

    return app.story()
  })

  it('beforeAppear', () => {
    const onBeforeAppear = jest.fn()
    const app = createApp({}, allProps({ onBeforeAppear }))

    console.log(app.tree())
    console.log(app.tree().children[0])

    expect(onBeforeAppear).toBeCalled()

    app.snap()

    return app.story()
  })

  it('enter', () => {
    const onEnter = jest.fn()
    const app = createApp({}, allProps({ onEnter }), { type: 'EMPTY' })

    app.snap({ type: 'FULL' })

    console.log(app.tree())
    console.log(app.tree().children[0])
    expect(onEnter).toBeCalled()

    return app.story()
  })

  it('beforeEnter', () => {
    const onBeforeEnter = jest.fn()
    const app = createApp({}, allProps({ onBeforeEnter }), { type: 'EMPTY' })

    app.snap({ type: 'FULL' })

    console.log(app.tree())
    console.log(app.tree().children[0])
    expect(onBeforeEnter).toBeCalled()

    return app.story()
  })

  it('leave', () => {
    const onLeave = jest.fn()
    const app = createApp({}, allProps({ onLeave }))

    app.snap({ type: 'EMPTY' })

    console.log(app.tree())
    expect(onLeave).toBeCalled()

    return app.story()
  })

  it('beforeLeave', () => {
    const onBeforeLeave = jest.fn()
    const app = createApp({}, allProps({ onBeforeLeave }))

    app.snap({ type: 'EMPTY' })

    console.log(app.tree())
    expect(onBeforeLeave).toBeCalled()

    return app.story()
  })
})
