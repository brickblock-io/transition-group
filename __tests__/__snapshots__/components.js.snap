// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Transition /> -- all props filled appear 1`] = `
Object {
  "className": "APPEAR",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<Transition /> -- all props filled appear 2`] = `
Object {
  "className": "APPEAR",
  "method": "setTimeoutAnimationFrame",
  "ms": 44,
}
`;

exports[`<Transition /> -- all props filled appear 3`] = `
Object {
  "className": "APPEAR-active",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<Transition /> -- all props filled appear 4`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 711,
}
`;

exports[`<Transition /> -- all props filled appear 5`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "horizontal",
}
`;

exports[`<Transition /> -- all props filled appear 6`] = `
<div>
  <div
    className="child"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
</div>
`;

exports[`<Transition /> -- all props filled beforeAppear 1`] = `
Object {
  "className": "APPEAR",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<Transition /> -- all props filled beforeAppear 2`] = `
Object {
  "className": "APPEAR",
  "method": "setTimeoutAnimationFrame",
  "ms": 44,
}
`;

exports[`<Transition /> -- all props filled beforeAppear 3`] = `
Object {
  "className": "APPEAR-active",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<Transition /> -- all props filled beforeAppear 4`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 711,
}
`;

exports[`<Transition /> -- all props filled beforeAppear 5`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "horizontal",
}
`;

exports[`<Transition /> -- all props filled beforeAppear 6`] = `
<div>
  <div
    className="child"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
</div>
`;

exports[`<Transition /> -- all props filled beforeEnter 1`] = `
Object {
  "type": "FULL",
}
`;

exports[`<Transition /> -- all props filled beforeEnter 2`] = `
Object {
  "className": "ENTER",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<Transition /> -- all props filled beforeEnter 3`] = `
Object {
  "className": "ENTER",
  "method": "setTimeoutAnimationFrame",
  "ms": 54,
}
`;

exports[`<Transition /> -- all props filled beforeEnter 4`] = `
Object {
  "className": "ENTER-active",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<Transition /> -- all props filled beforeEnter 5`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 722,
}
`;

exports[`<Transition /> -- all props filled beforeEnter 6`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "horizontal",
}
`;

exports[`<Transition /> -- all props filled beforeEnter 7`] = `
Object {
  "empty": false,
  "num": 0,
}
`;

exports[`<Transition /> -- all props filled beforeEnter 8`] = `
<div>
  <div
    className="child"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
</div>
`;

exports[`<Transition /> -- all props filled beforeLeave 1`] = `
Object {
  "className": "APPEAR",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<Transition /> -- all props filled beforeLeave 2`] = `
Object {
  "className": "APPEAR",
  "method": "setTimeoutAnimationFrame",
  "ms": 44,
}
`;

exports[`<Transition /> -- all props filled beforeLeave 3`] = `
Object {
  "className": "APPEAR-active",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<Transition /> -- all props filled beforeLeave 4`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 711,
}
`;

exports[`<Transition /> -- all props filled beforeLeave 5`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "horizontal",
}
`;

exports[`<Transition /> -- all props filled beforeLeave 6`] = `
Object {
  "type": "EMPTY",
}
`;

exports[`<Transition /> -- all props filled beforeLeave 7`] = `
Object {
  "className": "LEAVE",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<Transition /> -- all props filled beforeLeave 8`] = `
Object {
  "className": "LEAVE",
  "method": "setTimeoutAnimationFrame",
  "ms": 64,
}
`;

exports[`<Transition /> -- all props filled beforeLeave 9`] = `
Object {
  "className": "LEAVE-active",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<Transition /> -- all props filled beforeLeave 10`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 733,
}
`;

exports[`<Transition /> -- all props filled beforeLeave 11`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "horizontal",
}
`;

exports[`<Transition /> -- all props filled beforeLeave 12`] = `
Object {
  "empty": true,
  "num": 0,
}
`;

exports[`<Transition /> -- all props filled beforeLeave 13`] = `<div />`;

exports[`<Transition /> -- all props filled enter 1`] = `
Object {
  "type": "FULL",
}
`;

exports[`<Transition /> -- all props filled enter 2`] = `
Object {
  "className": "ENTER",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<Transition /> -- all props filled enter 3`] = `
Object {
  "className": "ENTER",
  "method": "setTimeoutAnimationFrame",
  "ms": 54,
}
`;

exports[`<Transition /> -- all props filled enter 4`] = `
Object {
  "className": "ENTER-active",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<Transition /> -- all props filled enter 5`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 722,
}
`;

exports[`<Transition /> -- all props filled enter 6`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "horizontal",
}
`;

exports[`<Transition /> -- all props filled enter 7`] = `
Object {
  "empty": false,
  "num": 0,
}
`;

exports[`<Transition /> -- all props filled enter 8`] = `
<div>
  <div
    className="child"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
</div>
`;

exports[`<Transition /> -- all props filled leave 1`] = `
Object {
  "className": "APPEAR",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<Transition /> -- all props filled leave 2`] = `
Object {
  "className": "APPEAR",
  "method": "setTimeoutAnimationFrame",
  "ms": 44,
}
`;

exports[`<Transition /> -- all props filled leave 3`] = `
Object {
  "className": "APPEAR-active",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<Transition /> -- all props filled leave 4`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 711,
}
`;

exports[`<Transition /> -- all props filled leave 5`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "horizontal",
}
`;

exports[`<Transition /> -- all props filled leave 6`] = `
Object {
  "type": "EMPTY",
}
`;

exports[`<Transition /> -- all props filled leave 7`] = `
Object {
  "className": "LEAVE",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<Transition /> -- all props filled leave 8`] = `
Object {
  "className": "LEAVE",
  "method": "setTimeoutAnimationFrame",
  "ms": 64,
}
`;

exports[`<Transition /> -- all props filled leave 9`] = `
Object {
  "className": "LEAVE-active",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<Transition /> -- all props filled leave 10`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 733,
}
`;

exports[`<Transition /> -- all props filled leave 11`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "horizontal",
}
`;

exports[`<Transition /> -- all props filled leave 12`] = `
Object {
  "empty": true,
  "num": 0,
}
`;

exports[`<Transition /> -- all props filled leave 13`] = `<div />`;

exports[`<TransitionGroup /> + <Transition /> -- no props appear 1`] = `
Object {
  "className": "appear",
  "method": "addClass",
  "name": "",
}
`;

exports[`<TransitionGroup /> + <Transition /> -- no props appear 2`] = `
Object {
  "className": "appear",
  "method": "setTimeoutAnimationFrame",
  "ms": 34,
}
`;

exports[`<TransitionGroup /> + <Transition /> -- no props appear 3`] = `
Object {
  "className": "appear-active",
  "method": "addClass",
  "name": "",
}
`;

exports[`<TransitionGroup /> + <Transition /> -- no props appear 4`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 35,
}
`;

exports[`<TransitionGroup /> + <Transition /> -- no props appear 5`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "",
}
`;

exports[`<TransitionGroup /> + <Transition /> -- no props appear 6`] = `
<div>
  <div
    className="child"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
</div>
`;

exports[`<TransitionGroup /> + <Transition /> -- no props enter + leave -- simple duration (debounced) 1`] = `
Object {
  "className": "appear",
  "method": "addClass",
  "name": "",
}
`;

exports[`<TransitionGroup /> + <Transition /> -- no props enter + leave -- simple duration (debounced) 2`] = `
Object {
  "className": "appear",
  "method": "setTimeoutAnimationFrame",
  "ms": 34,
}
`;

exports[`<TransitionGroup /> + <Transition /> -- no props enter + leave -- simple duration (debounced) 3`] = `
Object {
  "className": "appear-active",
  "method": "addClass",
  "name": "",
}
`;

exports[`<TransitionGroup /> + <Transition /> -- no props enter + leave -- simple duration (debounced) 4`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 535,
}
`;

exports[`<TransitionGroup /> + <Transition /> -- no props enter + leave -- simple duration (debounced) 5`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "",
}
`;

exports[`<TransitionGroup /> + <Transition /> -- no props enter + leave -- simple duration (debounced) 6`] = `
Object {
  "type": "SWITCH",
}
`;

exports[`<TransitionGroup /> + <Transition /> -- no props enter + leave -- simple duration (debounced) 7`] = `
Object {
  "empty": false,
  "num": 1,
}
`;

exports[`<TransitionGroup /> + <Transition /> -- no props enter + leave -- simple duration (debounced) 8`] = `
<div>
  <div
    className="child"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
</div>
`;

exports[`<TransitionGroup /> + <Transition /> -- no props enter + leave 1`] = `
Object {
  "className": "appear",
  "method": "addClass",
  "name": "",
}
`;

exports[`<TransitionGroup /> + <Transition /> -- no props enter + leave 2`] = `
Object {
  "className": "appear",
  "method": "setTimeoutAnimationFrame",
  "ms": 34,
}
`;

exports[`<TransitionGroup /> + <Transition /> -- no props enter + leave 3`] = `
Object {
  "className": "appear-active",
  "method": "addClass",
  "name": "",
}
`;

exports[`<TransitionGroup /> + <Transition /> -- no props enter + leave 4`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 35,
}
`;

exports[`<TransitionGroup /> + <Transition /> -- no props enter + leave 5`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "",
}
`;

exports[`<TransitionGroup /> + <Transition /> -- no props enter + leave 6`] = `
Object {
  "type": "SWITCH",
}
`;

exports[`<TransitionGroup /> + <Transition /> -- no props enter + leave 7`] = `
Object {
  "className": "enter",
  "method": "addClass",
  "name": "",
}
`;

exports[`<TransitionGroup /> + <Transition /> -- no props enter + leave 8`] = `
Object {
  "className": "enter",
  "method": "setTimeoutAnimationFrame",
  "ms": 34,
}
`;

exports[`<TransitionGroup /> + <Transition /> -- no props enter + leave 9`] = `
Object {
  "className": "enter-active",
  "method": "addClass",
  "name": "",
}
`;

exports[`<TransitionGroup /> + <Transition /> -- no props enter + leave 10`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 35,
}
`;

exports[`<TransitionGroup /> + <Transition /> -- no props enter + leave 11`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "",
}
`;

exports[`<TransitionGroup /> + <Transition /> -- no props enter + leave 12`] = `
Object {
  "className": "leave",
  "method": "addClass",
  "name": "",
}
`;

exports[`<TransitionGroup /> + <Transition /> -- no props enter + leave 13`] = `
Object {
  "className": "leave",
  "method": "setTimeoutAnimationFrame",
  "ms": 34,
}
`;

exports[`<TransitionGroup /> + <Transition /> -- no props enter + leave 14`] = `
Object {
  "className": "leave-active",
  "method": "addClass",
  "name": "",
}
`;

exports[`<TransitionGroup /> + <Transition /> -- no props enter + leave 15`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 35,
}
`;

exports[`<TransitionGroup /> + <Transition /> -- no props enter + leave 16`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "",
}
`;

exports[`<TransitionGroup /> + <Transition /> -- no props enter + leave 17`] = `
Object {
  "empty": false,
  "num": 1,
}
`;

exports[`<TransitionGroup /> + <Transition /> -- no props enter + leave 18`] = `
<div>
  <div
    className="child"
    style={
      Object {
        "backgroundColor": "purple",
      }
    }
  />
</div>
`;

exports[`<TransitionGroup /> -- all props filled appear 1`] = `
Object {
  "className": "APPEAR",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<TransitionGroup /> -- all props filled appear 2`] = `
Object {
  "className": "APPEAR",
  "method": "setTimeoutAnimationFrame",
  "ms": 44,
}
`;

exports[`<TransitionGroup /> -- all props filled appear 3`] = `
Object {
  "className": "APPEAR-active",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<TransitionGroup /> -- all props filled appear 4`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 711,
}
`;

exports[`<TransitionGroup /> -- all props filled appear 5`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "horizontal",
}
`;

exports[`<TransitionGroup /> -- all props filled appear 6`] = `
<div>
  <div
    className="child"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
</div>
`;

exports[`<TransitionGroup /> -- all props filled beforeAppear 1`] = `
Object {
  "className": "APPEAR",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<TransitionGroup /> -- all props filled beforeAppear 2`] = `
Object {
  "className": "APPEAR",
  "method": "setTimeoutAnimationFrame",
  "ms": 44,
}
`;

exports[`<TransitionGroup /> -- all props filled beforeAppear 3`] = `
Object {
  "className": "APPEAR-active",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<TransitionGroup /> -- all props filled beforeAppear 4`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 711,
}
`;

exports[`<TransitionGroup /> -- all props filled beforeAppear 5`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "horizontal",
}
`;

exports[`<TransitionGroup /> -- all props filled beforeAppear 6`] = `
<div>
  <div
    className="child"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
</div>
`;

exports[`<TransitionGroup /> -- all props filled beforeEnter 1`] = `
Object {
  "type": "FULL",
}
`;

exports[`<TransitionGroup /> -- all props filled beforeEnter 2`] = `
Object {
  "className": "ENTER",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<TransitionGroup /> -- all props filled beforeEnter 3`] = `
Object {
  "className": "ENTER",
  "method": "setTimeoutAnimationFrame",
  "ms": 54,
}
`;

exports[`<TransitionGroup /> -- all props filled beforeEnter 4`] = `
Object {
  "className": "ENTER-active",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<TransitionGroup /> -- all props filled beforeEnter 5`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 722,
}
`;

exports[`<TransitionGroup /> -- all props filled beforeEnter 6`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "horizontal",
}
`;

exports[`<TransitionGroup /> -- all props filled beforeEnter 7`] = `
Object {
  "empty": false,
  "num": 0,
}
`;

exports[`<TransitionGroup /> -- all props filled beforeEnter 8`] = `
<div>
  <div
    className="child"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
</div>
`;

exports[`<TransitionGroup /> -- all props filled beforeLeave 1`] = `
Object {
  "className": "APPEAR",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<TransitionGroup /> -- all props filled beforeLeave 2`] = `
Object {
  "className": "APPEAR",
  "method": "setTimeoutAnimationFrame",
  "ms": 44,
}
`;

exports[`<TransitionGroup /> -- all props filled beforeLeave 3`] = `
Object {
  "className": "APPEAR-active",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<TransitionGroup /> -- all props filled beforeLeave 4`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 711,
}
`;

exports[`<TransitionGroup /> -- all props filled beforeLeave 5`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "horizontal",
}
`;

exports[`<TransitionGroup /> -- all props filled beforeLeave 6`] = `
Object {
  "type": "EMPTY",
}
`;

exports[`<TransitionGroup /> -- all props filled beforeLeave 7`] = `
Object {
  "className": "LEAVE",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<TransitionGroup /> -- all props filled beforeLeave 8`] = `
Object {
  "className": "LEAVE",
  "method": "setTimeoutAnimationFrame",
  "ms": 64,
}
`;

exports[`<TransitionGroup /> -- all props filled beforeLeave 9`] = `
Object {
  "className": "LEAVE-active",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<TransitionGroup /> -- all props filled beforeLeave 10`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 733,
}
`;

exports[`<TransitionGroup /> -- all props filled beforeLeave 11`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "horizontal",
}
`;

exports[`<TransitionGroup /> -- all props filled beforeLeave 12`] = `
Object {
  "empty": true,
  "num": 0,
}
`;

exports[`<TransitionGroup /> -- all props filled beforeLeave 13`] = `<div />`;

exports[`<TransitionGroup /> -- all props filled enter 1`] = `
Object {
  "type": "FULL",
}
`;

exports[`<TransitionGroup /> -- all props filled enter 2`] = `
Object {
  "className": "ENTER",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<TransitionGroup /> -- all props filled enter 3`] = `
Object {
  "className": "ENTER",
  "method": "setTimeoutAnimationFrame",
  "ms": 54,
}
`;

exports[`<TransitionGroup /> -- all props filled enter 4`] = `
Object {
  "className": "ENTER-active",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<TransitionGroup /> -- all props filled enter 5`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 722,
}
`;

exports[`<TransitionGroup /> -- all props filled enter 6`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "horizontal",
}
`;

exports[`<TransitionGroup /> -- all props filled enter 7`] = `
Object {
  "empty": false,
  "num": 0,
}
`;

exports[`<TransitionGroup /> -- all props filled enter 8`] = `
<div>
  <div
    className="child"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
</div>
`;

exports[`<TransitionGroup /> -- all props filled leave 1`] = `
Object {
  "className": "APPEAR",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<TransitionGroup /> -- all props filled leave 2`] = `
Object {
  "className": "APPEAR",
  "method": "setTimeoutAnimationFrame",
  "ms": 44,
}
`;

exports[`<TransitionGroup /> -- all props filled leave 3`] = `
Object {
  "className": "APPEAR-active",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<TransitionGroup /> -- all props filled leave 4`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 711,
}
`;

exports[`<TransitionGroup /> -- all props filled leave 5`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "horizontal",
}
`;

exports[`<TransitionGroup /> -- all props filled leave 6`] = `
Object {
  "type": "EMPTY",
}
`;

exports[`<TransitionGroup /> -- all props filled leave 7`] = `
Object {
  "className": "LEAVE",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<TransitionGroup /> -- all props filled leave 8`] = `
Object {
  "className": "LEAVE",
  "method": "setTimeoutAnimationFrame",
  "ms": 64,
}
`;

exports[`<TransitionGroup /> -- all props filled leave 9`] = `
Object {
  "className": "LEAVE-active",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<TransitionGroup /> -- all props filled leave 10`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 733,
}
`;

exports[`<TransitionGroup /> -- all props filled leave 11`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "horizontal",
}
`;

exports[`<TransitionGroup /> -- all props filled leave 12`] = `
Object {
  "empty": true,
  "num": 0,
}
`;

exports[`<TransitionGroup /> -- all props filled leave 13`] = `<div />`;

exports[`<TransitionGroup /> -- all props filled onEmpty callback 1`] = `
Object {
  "className": "appear",
  "method": "addClass",
  "name": "",
}
`;

exports[`<TransitionGroup /> -- all props filled onEmpty callback 2`] = `
Object {
  "className": "appear",
  "method": "setTimeoutAnimationFrame",
  "ms": 34,
}
`;

exports[`<TransitionGroup /> -- all props filled onEmpty callback 3`] = `
Object {
  "className": "appear-active",
  "method": "addClass",
  "name": "",
}
`;

exports[`<TransitionGroup /> -- all props filled onEmpty callback 4`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 35,
}
`;

exports[`<TransitionGroup /> -- all props filled onEmpty callback 5`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "",
}
`;

exports[`<TransitionGroup /> -- all props filled onEmpty callback 6`] = `
Object {
  "type": "EMPTY",
}
`;

exports[`<TransitionGroup /> -- all props filled onEmpty callback 7`] = `
Object {
  "className": undefined,
  "method": "setTimeoutAnimationFrame",
  "ms": 0,
}
`;

exports[`<TransitionGroup /> -- all props filled onEmpty callback 8`] = `
Object {
  "className": "leave",
  "method": "addClass",
  "name": "",
}
`;

exports[`<TransitionGroup /> -- all props filled onEmpty callback 9`] = `
Object {
  "className": "leave",
  "method": "setTimeoutAnimationFrame",
  "ms": 34,
}
`;

exports[`<TransitionGroup /> -- all props filled onEmpty callback 10`] = `
Object {
  "className": "leave-active",
  "method": "addClass",
  "name": "",
}
`;

exports[`<TransitionGroup /> -- all props filled onEmpty callback 11`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 35,
}
`;

exports[`<TransitionGroup /> -- all props filled onEmpty callback 12`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "",
}
`;

exports[`<TransitionGroup /> -- all props filled onEmpty callback 13`] = `
Object {
  "empty": true,
  "num": 0,
}
`;

exports[`<TransitionGroup /> -- all props filled onEmpty callback 14`] = `<div />`;

exports[`<TransitionGroup /> -- all props filled onFull callback 1`] = `
Object {
  "type": "FULL",
}
`;

exports[`<TransitionGroup /> -- all props filled onFull callback 2`] = `
Object {
  "className": undefined,
  "method": "setTimeoutAnimationFrame",
  "ms": 0,
}
`;

exports[`<TransitionGroup /> -- all props filled onFull callback 3`] = `
Object {
  "className": "enter",
  "method": "addClass",
  "name": "",
}
`;

exports[`<TransitionGroup /> -- all props filled onFull callback 4`] = `
Object {
  "className": "enter",
  "method": "setTimeoutAnimationFrame",
  "ms": 34,
}
`;

exports[`<TransitionGroup /> -- all props filled onFull callback 5`] = `
Object {
  "className": "enter-active",
  "method": "addClass",
  "name": "",
}
`;

exports[`<TransitionGroup /> -- all props filled onFull callback 6`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 35,
}
`;

exports[`<TransitionGroup /> -- all props filled onFull callback 7`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "",
}
`;

exports[`<TransitionGroup /> -- all props filled onFull callback 8`] = `
Object {
  "empty": false,
  "num": 0,
}
`;

exports[`<TransitionGroup /> -- all props filled onFull callback 9`] = `
<div>
  <div
    className="child"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
</div>
`;
